### ISSUE TITLE
Loop in gsub

### ISSUE THREAD

**Issue by rjmac** [myfreeweb/pcre-heavy#2]
`gsub`, when replacing, offsets after each replacement by the length of the match, not the length of the replacement, which means that if the replacement contains something that will re-match after that limit, gsub will loop forever:

``` haskell
λ> gsub [re|bad|] "xxbad" "this is bad, right?" -- length "xx" < length "bad", so ok
"this is xxbad, right?"
λ> gsub [re|bad|] "xxxbad" "this is bad, right?" -- length "xxx" >= length "bad", so loop
"^CInterrupted.
```

Similarly, it can miss targets that occur soon after a shortening replacement:

``` haskell
λ> gsub [re|good|] "bad" "goodgoodgood"
"badgoodbad"
```

The ultimate bug's in `rawSub`, where it returns `end` instead of `offset + length replacement`.

**Comment by valpackett** [comment]
Oh fuck. Thanks for noticing! Going to fix right now

**Comment by valpackett** [comment]
Released 0.2.3!

P.S. it was `begin + length replacement`, not `offset + length replacement` :-)
